### DISCUSSION TITLE
Looping video

### DISCUSSION THREAD

**Discussion by kir0ul** [quarto-dev/quarto-cli#3060] [Q&A]
Hi,

Thanks for this super useful tool!

I was wondering if there's a way to add a `loop="true"` attribute to a `<video>` element in a `reveal.js` presentation?

**Comment by mcanouil** [comment]
I am not sure to understand your issue.
Reveal.js is HTML, so everything possible in HTML (which includes make a video start over in a loop) is possible the exact same way, although you myght be mixing two syntax, the one for `<embed loop="true">` and `<video loop>` (<https://www.w3schools.com/TAgs/att_video_loop.asp>).
There is also a special syntax for YouTube videos (see <https://www.w3schools.com/html/html_youtube.asp>).

**Reply by kir0ul** [reply]
Sorry if that's not clear, my question is that I have a local MP4 video like below, and I'd like it to loop over, how can I do that with the Quarto syntax?
```
{{< video medias/my_video.mp4 width="400" height="300" >}}
```

**Reply by mcanouil** [reply]
Oh, you meant in the video shortcode.

Short answer: no, you can't (see <https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/extensions/quarto/video/video.lua>).
You have to write yourself the proper HTML or use browser developer mode to get the created HTML, copy/paste it back in your document, then modify it to your need.

**Reply by JekelyLab** [reply]
it works if you write`![](medias/my_video.mp4){.absolute left="0" top="250" height="250" loop="true"}`

**Reply by mcanouil** [reply]
Yes, as it does with plain HTML.
Thanks for pointing this.

Note: it does not change the fact that the shortcode still can't handle this.

**Reply by danielvartan** [reply]
@JekelyLab, your solution didn’t work for me. Here’s an HTML alternative in case anyone else is looking.

```html
<video width="90%" style="display: block; margin: 0 auto;" controls muted autoplay loop>
  <source src="video.webm" type="video/webm">
</video>
```

**Reply by JekelyLab** [reply]
Thanks, this works. However, I couldn't combine it with the columns syntax: 

````
::: columns
::: {.column width="40%"}
<video width="50%" style="display: block; margin: 0 auto;" controls muted autoplay loop>
  <source src="video1.webm" type="video/webm">
</video>
:::
::: {.column width="60%"}
<video width="50%" style="display: block; margin: 0 auto;" controls muted autoplay loop>
  <source src="video2.webm" type="video/webm">
</video>
:::
:::
````

**Reply by mcanouil** [reply]
@JekelyLab 
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

**Reply by mcanouil** [reply]
A Quarto document should never contain bare HTML (or LaTeX, or Typst, etc.) code.

See [Markdown basics - raw content](https://quarto.org/docs/authoring/markdown-basics.html#raw-content).

Note also that by default, elements are relative to each other, so you are setting your videos to be half the width of the container.

**Reply by mcanouil** [reply]
````qmd
---
title: "Testing"
format: html
engine: markdown
---

::: {.columns}

::: {.column width="40%"}

```{=html}
<video width="50%" style="display: block; margin: 0 auto;" controls muted autoplay loop>
  <source src="video.webm" type="video/webm">
</video>
```

:::

::: {.column width="60%"}

```{=html}
<video width="50%" style="display: block; margin: 0 auto;" controls muted autoplay loop>
  <source src="video.webm" type="video/webm">
</video>
```

:::

:::
````


https://github.com/user-attachments/assets/852354ff-4fde-4550-801b-0d8e0055794a

**Comment by cderv** [comment]
This is now tracked in https://github.com/quarto-dev/quarto-cli/issues/3210 as a feature request
